File: tests/nodejs/cases/case_251_closure_registry_delay_05.js
/**
 * Task case 251: closure_registry_delay_05 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case251ClosureRegistryDelay05(input) {
  const taskName = 'closure_registry_delay_05';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const later = () => { retainedAudit.push(payload); return payload.input; };
  // ESCAPE: retained closure executes after caller returns.
  retainedHandlers.push(later);
  return 'ok';
}

module.exports = { case251ClosureRegistryDelay05 };

File: tests/go/cases/case_251_closure_registry_delay_05.go
package escape_tests

var retainedCase251 = []map[string]string{}

func Case251ClosureRegistryDelay05(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "closure_registry_delay_05",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "closure_registry_delay_05:" + raw,
	}
	handler := func() string { retainedCase251 = append(retainedCase251, payload); return payload["input"] }
	_ = handler
	// ESCAPE: retained closure-like dispatch.
	retainedCase251 = append(retainedCase251, map[string]string{"h": payload["task"]})
	return "ok"
}
